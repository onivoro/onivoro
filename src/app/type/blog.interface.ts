import { IPage } from "./page.interface";
import { ISection } from "./section.interface";

export interface IBlog extends IPage {
    sections: ISection[];
}