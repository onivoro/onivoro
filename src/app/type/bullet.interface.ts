export interface IBullet {
    icon: string;
    text: string;
}