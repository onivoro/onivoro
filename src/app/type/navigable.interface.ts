export interface INavigable {
    label: string;
    icon: string;
    slug: string;
}