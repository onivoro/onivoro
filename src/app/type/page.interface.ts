export interface IPage {
    title: string;
    metatags?: any;
    structuredData?: any;
    lastUpdatedAt?: string;
    lastUpdatedBy?: string;
    createdAt?: string;
    createdBy?: string;
}