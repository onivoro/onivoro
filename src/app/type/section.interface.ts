import { IParagraph } from './paragraph.interface';

export interface ISection {
    heading: string;
    paragraphs: IParagraph[];

}