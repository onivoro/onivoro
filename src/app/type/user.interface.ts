export interface IUser {
    isAdmin?: boolean;
    email?: string;
    phone?: number;
    name?: string;
}