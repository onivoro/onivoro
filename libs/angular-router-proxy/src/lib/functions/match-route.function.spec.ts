import { TRouteEnhanced } from '../types/route-enhanced.type';
import { matchRoute } from './match-route.function';
import { enhanceRoutes } from './enhance-routes.function';

const routesEnhanced: TRouteEnhanced[] = enhanceRoutes([
    { path: 'items/:id' },
    { path: '' },
]);

describe('matchRoute', () => {
    it.each([
        [['items/',],],
        [['items',],],
        [['items', '2345'],],
        [['', 'items', '2345'],],
        [['/'],],
        [[''],],
    ])('parses url into segments and param map given url %j', (segments: string[]) => {
        expect(matchRoute(segments, routesEnhanced)).toMatchSnapshot();
    });
});