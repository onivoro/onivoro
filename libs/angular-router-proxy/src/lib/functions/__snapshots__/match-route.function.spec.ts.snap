// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matchRoute parses url into segments and param map given url ["","items","2345"] 1`] = `
Object {
  "params": Object {
    "id": "2345",
  },
  "route": Object {
    "params": Object {
      ":id": "",
    },
    "path": "items/:id",
    "segments": Array [
      "items",
      ":id",
    ],
  },
}
`;

exports[`matchRoute parses url into segments and param map given url [""] 1`] = `
Object {
  "params": Object {},
  "route": undefined,
}
`;

exports[`matchRoute parses url into segments and param map given url ["/"] 1`] = `
Object {
  "params": Object {},
  "route": Object {
    "params": Object {},
    "path": "",
    "segments": Array [
      "",
    ],
  },
}
`;

exports[`matchRoute parses url into segments and param map given url ["items","2345"] 1`] = `
Object {
  "params": Object {
    "id": "2345",
  },
  "route": Object {
    "params": Object {
      ":id": "",
    },
    "path": "items/:id",
    "segments": Array [
      "items",
      ":id",
    ],
  },
}
`;

exports[`matchRoute parses url into segments and param map given url ["items"] 1`] = `
Object {
  "params": Object {},
  "route": undefined,
}
`;

exports[`matchRoute parses url into segments and param map given url ["items/"] 1`] = `
Object {
  "params": Object {},
  "route": undefined,
}
`;
